pkg/sql/logictest/tests/5node/5node_test: TestLogic_distsql_stats failed

pkg/sql/logictest/tests/5node/5node_test.TestLogic_distsql_stats [failed](https://tanzanite.cluster.engflow.com/invocations/default/83d7e821-4089-4151-8d9e-9e83eb9ea4ca?testReportRun=18&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvc3FsL2xvZ2ljdGVzdC90ZXN0cy81bm9kZTo1bm9kZV90ZXN0) on release-24.1 @ [440813cd94435de6806c569497ac5e2f06826076](https://github.com/cockroachdb/cockroach/commits/440813cd94435de6806c569497ac5e2f06826076):


```
    logic.go:3072: let $d_desc_null_stats = [{"avg_size": 2, "columns": ["a"], "created_at": "2025-02-02 10:14:04.634671", "distinct_count": 7, "histo_buckets": [{"distinct_range": 0, "num_eq": 1, "num_range": 0, "upper_bound": "0"}, {"distinct_range": 0, "num_eq": 1, "num_range": 0, "upper_bound": "1"}, {"distinct_range": 0, "num_eq": 1, "num_range": 0, "upper_bound": "2"}, {"distinct_range": 0, "num_eq": 1, "num_range": 0, "upper_bound": "3"}, {"distinct_range": 0, "num_eq": 1, "num_range": 0, "upper_bound": "4"}, {"distinct_range": 0, "num_eq": 1, "num_range": 0, "upper_bound": "5"}], "histo_col_type": "INT8", "histo_version": 3, "name": "sdn", "null_count": 2, "row_count": 8}]
    logic.go:3072: let $s_stats = [{"avg_size": 3, "columns": ["s"], "created_at": "2025-02-02 10:14:04.85218", "distinct_count": 3, "histo_buckets": [{"distinct_range": 0, "num_eq": 1, "num_range": 0, "upper_bound": "c"}, {"distinct_range": 0, "num_eq": 1, "num_range": 0, "upper_bound": "d"}, {"distinct_range": 0, "num_eq": 1, "num_range": 0, "upper_bound": "e"}], "histo_col_type": "STRING", "histo_version": 3, "name": "str", "null_count": 0, "row_count": 3}]
    logic.go:3072: let $j_stats = [{"avg_size": 18, "columns": ["j"], "created_at": "2025-02-02 10:14:05.010009", "distinct_count": 3, "histo_buckets": [{"distinct_range": 0, "num_eq": 1, "num_range": 0, "upper_bound": "{\"1\": \"10\"}"}, {"distinct_range": 0, "num_eq": 1, "num_range": 0, "upper_bound": "{\"2\": \"20\"}"}, {"distinct_range": 0, "num_eq": 1, "num_range": 0, "upper_bound": "{\"3\": \"30\"}"}], "histo_col_type": "JSONB", "histo_version": 3, "name": "j_full", "null_count": 0, "row_count": 3}]
    logic.go:3072: let $only_null_stat = [{"avg_size": 0, "columns": ["a"], "created_at": "2025-02-02 10:14:05.257623", "distinct_count": 1, "histo_col_type": "INT8", "histo_version": 3, "name": "only_null_stat", "null_count": 3, "row_count": 3}]
    logic.go:3072: let $hist_id_1 = 1043401426979848197
    logic.go:3072: let $hist_d = 1043401428135313413
    logic.go:3072: let $hist_e = 1043401428313047045
    logic.go:3072: let $hist_f = 1043401428574830597
[10:14:06] --- progress: /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node/5node_test_/5node_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_stats: 340 statements
    logic.go:3072: let $hist_crdb_internal_idx_expr = 1043401429648375813
    logic.go:3072: let $hist_d = 1043401429817851909
    logic.go:3072: let $hist_e = 1043401429979660293
    logic.go:3072: let $hist_crdb_internal_idx_expr = 1043401430122758149
[10:14:20] --- progress: /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node/5node_test_/5node_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_stats: 357 statements
    logic.go:3072: let $job = 1043401430288596993
[10:14:24] --- progress: /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node/5node_test_/5node_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_stats: 362 statements
    logic.go:3072: let $hist_status = 1043401490900910085
    logic.go:3026: 
         
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node/5node_test_/5node_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_stats:2983: EXPLAIN SELECT count(*) FROM t138809 WHERE b > 1
        expected:
            distribution: full
            vectorized: true
            ·
            • group (scalar)
            │ estimated row count: 1
            │
            └── • scan
                  estimated row count: 2 (100% of the table; stats collected <hidden> ago)
                  table: t138809@t138809_pkey
                  spans: [/1 - ]
            
        but found (query options: "") :
            distribution: full
            vectorized: true
            ·
            • group (scalar)
            │
            └── • scan
                  missing stats
                  table: t138809@t138809_pkey
                  spans: [/1 - ]
            
[10:14:25] --- done: /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node/5node_test_/5node_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_stats with config 5node: 384 tests, 1 failures
    logic.go:4201: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node/5node_test_/5node_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_stats:2997: error while processing
    logic.go:4201: /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node/5node_test_/5node_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_stats:2997: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLogic_distsql_stats1433256769
--- FAIL: TestLogic_distsql_stats (33.17s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=18</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139895 pkg/sql/logictest/tests/5node/5node_test: TestLogic_distsql_stats failed [C-test-failure O-robot T-sql-queries branch-release-25.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_distsql_stats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-47120

--- a/pkg/sql/logictest/testdata/logic_test/distsql_stats
+++ b/pkg/sql/logictest/testdata/logic_test/distsql_stats
@@ -3531,7 +3531,7 @@ column_names  row_count  null_count  distinct_count  avg_size  has_histogram
 {b}           2          0           2               8         true
 {c}           2          0           2               8         true
 
-query T
+query T retry
 EXPLAIN SELECT count(*) FROM t138809 WHERE b > 1
 ----
 distribution: full

